client/packages/common/src/ui/components/steppers/HorizontalStepper/HorizontalStepper.tsx

```diff
@@ -0,0 +1,231 @@
+    }
+
+    const style = {
+      border: 'solid',
+      borderWidth: '3px',
+      width: '30px',
+      height: '30px',
+      borderRadius: '30px',
+      textAlign: 'center' as 'center' | 'left' | 'right',
+      fontWeight: 700,
+      animation: active ? 'pulse 2s infinite' : 'none',
+      ...colors,
+    };
+
+    return <div style={style}>{icon}</div>;
+  };
+
+const getAnimationStyles = (colour: StepperColour) => {
+  const colour1 =
+    colour === 'primary' ? 'rgb(254, 147, 24, 0.4)' : 'rgba(53, 104, 212, 0.4)';
+  const colour2 = 'rgba(53, 104, 212, 0)';
+
+  return {
+    '@-webkit-keyframes pulse': {
+      '0%': {
+        '-webkit-box-shadow': `0 0 0 0 ${colour1}`,
+      },
+      '70%': {
+        '-webkit-box-shadow': `0 0 0 10px ${colour2}`,
+      },
+      '100%': {
+        '-webkit-box-shadow': `0 0 0 0 ${colour2}`,
+      },
+    },
+    '@keyframes pulse': {
+      '0%': {
+        '-moz-box-shadow': `0 0 0 0 ${colour1}`,
+        boxShadow: `0 0 0 0 ${colour1}`,
+      },
+      '70%': {
+        '-moz-box-shadow': `0 0 0 10px ${colour2}`,
+        boxShadow: `0 0 0 10px ${colour2}`,
+      },
+      '100%': {
+        '-moz-box-shadow': `0 0 0 0 ${colour2}`,
+        boxShadow: `0 0 0 0 ${colour2}`,
+      },
+    },
+  };
+};
+
+const getLabelStyle = (
+  colour: StepperColour,
+  completed?: boolean,
+  error?: boolean
+) => {
+  switch (true) {
+    case !!error:
+      return { color: 'error.main' };
+    case !!completed:
+      return { color: `${colour}.dark` };
+    default:
+      return { color: `${colour}.light` };
+  }
+};
+
+const usePaletteColour = (colour: StepperColour): PaletteColour => {
+  const theme = useTheme();
+  switch (colour) {
+    case 'secondary':
+      return {
+        light: theme.palette.gray.pale,
+        main: theme.palette.secondary.light,
+        dark: theme.palette.secondary.dark,
+        error: theme.palette.error.main,
+      };
+    default:
+      return {
+        light: theme.palette.primary.light,
+        main: theme.palette.primary.main,
+        dark: theme.palette.primary.dark,
+        error: theme.palette.error.main,
+      };
+  }
+};
+
+/* alternativeLabel shows icons on top */
+export const HorizontalStepper: FC<StepperProps> = ({
+  colour = 'secondary',
+  steps,
+}) => {
+  const paletteColour = usePaletteColour(colour);
+  const StyledConnector = getConnector(paletteColour);
+
+  return (
+    <Box flex={1}>
+      <GlobalStyles styles={getAnimationStyles(colour)} />
+      <Stepper
+        connector={<StyledConnector />}
+        activeStep={steps.findIndex(step => step?.active)}
+        orientation="horizontal"
+        alternativeLabel
+      >
+        {steps.map(step => {
+          const { active, completed, label, optional, icon, error } = step;
+          // There is no accessability role that I can find to accurately describe
+          // a stepper, so turning to testids to mark the active/completed steps
+          // for tests
```

Comment:
I couldn't quite see, are there tests for this ? (maybe in that case a method to extract test_id from test, i like the high level of abstraction here)

```
 <Step
              data-testid={getTestId(step)}
              key={label}
              active={active}
              completed={completed}
            >
```

Reply:
oops. thanks!
have added tests and stories for the new component.